sys/linux/auto.txt

```diff
@@ -6495,12 +8598,68 @@ udp_timeout_nla_policy$auto_nf_conntrack_proto_udp [
 	CTA_TIMEOUT_UDP_REPLIED		nlattr[CTA_TIMEOUT_UDP_REPLIED, int32]
 ] [varlen]
 
+unewset$auto_record {
+	sig	array[int32, 1]
+}
+
+uprobe_multi$auto_record {
+	path		intptr
+	offsets		intptr
+	ref_ctr_offsets	intptr
+	cookies		intptr
+	cnt		int32
+	flags		int32
+	pid		int32
+}
+
+user_desc$auto_record {
+	entry_number	int32
+	base_addr	int32
+	limit		int32
+	seg_32bit	int32
+	contents	int32
+	read_exec_only	int32
+	limit_in_pages	int32
+	seg_not_present	int32
+	useable		int32
+	lm		int32
+}
+
+user_mask$auto_record {
+	sig	array[int32, 1]
+}
+
+user_msghdr$auto_record {
+	msg_name	ptr[inout, void]
+	msg_namelen	int32
+	msg_iov		ptr[inout, iovec$auto_record]
+	msg_iovlen	int32
+	msg_control	ptr[inout, void]
+	msg_controllen	int32
+	msg_flags	int32
+}
+
 userspace_policy$auto_flow_netlink [
 	OVS_USERSPACE_ATTR_PID			nlattr[OVS_USERSPACE_ATTR_PID, int32]
 	OVS_USERSPACE_ATTR_USERDATA		nlattr[OVS_USERSPACE_ATTR_USERDATA, array[int8]]
 	OVS_USERSPACE_ATTR_EGRESS_TUN_PORT	nlattr[OVS_USERSPACE_ATTR_EGRESS_TUN_PORT, int32]
 ] [varlen]
 
+uset$auto_record {
+	sig	array[int32, 1]
```

Comment:
It looks like we have 9 or so copies of sigset_t struct under different names.
Did you use field name instead of struct name because of conflicts?
I wonder it can somehow have 1 sigset_t struct.

Reply:
I think the reason this happened is that it's an anonymous struct that's typedef'd so it just gets the field name as a name.

Reply:
I see. Please add a TODO that we should support typedefs of anon structs (they are very common in the kernel code).

Reply:
It's handled properly now.